src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -107,10 +107,24 @@
   <PropertyGroup>
     <RuntimeConfigName Condition="'$(BuildingNETCoreAppVertical)' == 'true'">netcoreapp.runtimeconfig.json</RuntimeConfigName>
     <RuntimeConfigName Condition="'$(BuildingNETFxVertical)' == 'true'">DesktopRunnerConfigFile.config</RuntimeConfigName>
+    <RuntimeConfigExtension Condition="'$(BuildingNETCoreAppVertical)' == 'true'">.runtimeconfig.json</RuntimeConfigExtension>
+    <RuntimeConfigExtension Condition="'$(BuildingNETFxVertical)' == 'true'">.exe.config</RuntimeConfigExtension>
     <RuntimeConfigPath Condition="'$(BuildingNETCoreAppVertical)' == 'true' AND '$(RuntimeConfigPath)' == '' AND '$(CodeCoverageEnabled)' == 'true' AND '$(UseCoverageDedicatedRuntime)' == 'true'">$(ToolsDir)coverage/$(RuntimeConfigName)</RuntimeConfigPath>
     <RuntimeConfigPath Condition="'$(RuntimeConfigPath)' == ''">$(ToolsDir)$(RuntimeConfigName)</RuntimeConfigPath>
+    
+    <!-- Copy the app.config / runtimeSettings.json file into the OutDir if the project is an executable test project. -->
+    <CopyConfigurationFiles Condition="'$(CopyConfigurationFiles)' == ''
+                                  AND ('$(BuildingNETCoreAppVertical)' == 'true' OR '$(BuildingNETFxVertical)' == 'true')
+                                  AND '$(IsTestProject)' == 'true'
+                                  AND '$(IsInvokableTestProject)' != 'true'">true</CopyConfigurationFiles>
   </PropertyGroup>
 
+  <Target Name="CopyConfigurationFilesTarget" Condition="'$(CopyConfigurationFiles)' == 'true'">
+    <Copy SourceFiles="$(RuntimeConfigPath)"
```

Comment:
Why use copy and not add it SuplementalTestData? I think it is safer to use the current binplace system rather than copying it manually to avoid race conditions. 

You could potentially make CopySupplementalTestData depend on this target.

Reply:
I tried different approaches with SupplementalTestData but it didn't work out. In fact it should just be part of the CopyToOutputDirectory items collection as we did included the configuration files manually in the respective projects in corefx. Added a commit to address that.

Reply:
The SupplementalTestData group only works for the invokable test projects. If you use that for non invokable test projects like the RemoteExecutorConsoleApp then you end up putting the test data into its output directory which then not gets binplaced into the invokable test project directory as we only binplace the executable and pdb files. I added it to the CopyToOutputItems as they will correctly be copied to the invokable test project folder.